Strassen Matrix Multiplication/Matrix.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  Matrix.swift
+//  
+//
+//  Created by Richard Ash on 10/28/16.
+//
+//
+
+import Foundation
+
+enum RowOrColumn {
```

Comment:
Since you're only using this internally for the `subscript`, it feels a bit better to put it in the `Matrix` namespace.